massa-execution-worker/src/controller.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+//! This module implements an execution controller.
+//! See massa-execution-exports/controller_traits.rs for functional details.
+
+use crate::execution::ExecutionState;
+use massa_execution_exports::{
+    ExecutionConfig, ExecutionController, ExecutionError, ExecutionManager, ExecutionOutput,
+    ReadOnlyExecutionRequest,
+};
+use massa_ledger::LedgerEntry;
+use massa_models::output_event::SCOutputEvent;
+use massa_models::prehash::Map;
+use massa_models::Address;
+use massa_models::OperationId;
+use massa_models::{Block, BlockId, Slot};
+use std::collections::{HashMap, VecDeque};
+use std::sync::{mpsc, Arc, Condvar, Mutex, RwLock};
```

Comment:
I think it would be better to use [`parking_lot`](https://docs.rs/crate/parking_lot/latest) for the condvar and locks, we'll be using it for storage as well.

Reply:
done